src/coordination/coordinator_instance.cpp

```diff
@@ -446,12 +526,30 @@ auto CoordinatorInstance::ForceResetCluster_() -> ForceResetClusterStateStatus {
     return !raft_state_->AppendSetInstanceAsReplicaLog(instance.InstanceName());
   };
   if (std::ranges::any_of(alive_instances, demote_to_replica_failed)) {
-    spdlog::error("Failed to send raft log or rpc that instance is demoted to replica.");
-    return ForceResetClusterStateStatus::RPC_FAILED;
+    spdlog::error(
+        "Exiting VerifyOrCorrectClusterState. Failed to send raft log or rpc that instance is demoted to replica.");
+    return VerifyOrCorrectClusterStateStatus::FAIL;
   }
 
+  auto const all_alive_replicas = std::ranges::all_of(alive_instances, [this](auto const &instance) {
+    if (raft_state_->HasReplicaState(instance.InstanceName())) {
+      return true;
+    }
+    spdlog::trace("Instance {} is not replica", instance.InstanceName());
+    return false;
+  });
+
+  MG_ASSERT(all_alive_replicas, "All instances should be replicas");
+
```

Comment:
Is this necessary? We are retrieving this info from raft_state, I don't think we need here. We could be interested into response from data instance but we already have this response from demotion

Reply:
will remove it